patch_recurrence/tools.py

```diff
@@ -46,3 +49,65 @@ def show_pair(pair):
 
     # show the images
     plt.show()
+
+
+'''
+Assigning colors to each patch based on bucket number - also, resizing img
+'''
+def set_buckets(img, patch, constants):
+    num_buckets = constants.NUM_BUCKETS
+    patch_size = constants.PATCH_SIZE
+
+    new_height = img.shape[0] - patch_size
+    new_width = img.shape[1] - patch_size
+    img = img[0:new_height, 0:new_width]
+    img = np.reshape(img, [-1, 3])
+
+    bucket_colors = []
+    l1 = [(0, 0, i*2 / num_buckets) for i in range(int(num_buckets/2))]
+    l2 = [(i*2 / num_buckets, i*2 / num_buckets, 1) for i in range(int(num_buckets/2))]
+    bucket_colors = list(reversed(l1 + l2))
+
+    for i in range(new_height*new_width):
```

Comment:
**nit** - space between binary operators, space after `,`. Install `flake8` in Sublime Text